wikilabels/wsgi/static/js/wikiLabels/Workspace.js

```
@@ -0,0 +1,338 @@
+
+		this.i = i;
+		this.selected = $.Callbacks();
+		this.activated = $.Callbacks();
+
+		this.load(taskData);
+	};
+	Task.prototype.handleClick = function (e) {
+		if ( !this.disable() ) {
+			this.activated.fire(this);
+		}
+	};
+	Task.prototype.load = function (taskData) {
+		this.$element.empty();
+		this.id = taskData.id;
+		this.data = taskData['task_data'];
+		this.label = new Label(taskData['labels'][0]);
+		this.$element.append(this.label.$element);
+	};
+	Task.prototype.select = function (selected) {
+		if ( selected === undefined) {
+			return this.$element.hasClass('selected');
+		} else if ( selected ) {
+			this.$element.addClass('selected');
+			this.selected.fire();
+			return this;
+		} else {
+			this.$element.removeClass('selected');
+			return this;
+		}
+	};
+	Task.prototype.setWidth = function (width) {
+		this.$element.css('width', width);
+	};
+	Task.prototype.disable = function (disabled) {
+		if ( disabled === undefined) {
+			return this.$element.hasClass('disabled');
+		} else if ( disabled ) {
+			this.$element.addClass('disabled');
+			return this;
+		} else {
+			this.$element.removeClass('disabled');
+			return this;
+		}
+	};
+	Task.prototype.complete = function () {
+		return this.label.complete();
+	};
+
+	var Label = function (labelData) {
+		this.$element = $("<div>").addClass("label");
+		this.timestamp = null;
+		this.data = null;
+
+		this.load(labelData);
+	};
+	Label.prototype.load = function (labelData) {
+		labelData = labelData || {};
+		this.timestamp = labelData['timestamp'];
+		this.data = labelData['data'];
+		this.complete(this.data !== undefined && this.data !== null);
+	};
+	Label.prototype.complete = function (completed) {
+		if ( completed === undefined) {
+			return this.$element.hasClass('completed');
+		} else if ( completed ) {
+			this.$element.addClass('completed');
+			return this;
+		} else {
+			this.$element.removeClass('completed');
+			return this;
+		}
```

I just looked at [mediawiki.user.js](https://gerrit.wikimedia.org/r/#/c/189675/2/resources/src/mediawiki/mediawiki.user.js) and they are solving a different problem.  So the two **if**s before the return are checking the state before trying to return a value from that function.  We follow a similar pattern at the beginning of [Home.js](https://github.com/halfak/Wiki-Labels/pull/6/files#diff-256522e5f6deba78b968ba5a842c044eR7), but we throw exceptions rather than return null and false(!?!?).